integration-kinesis-to-lambda-sample-code/Function.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.KinesisEvents;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace KinesisIntegrationSampleCode;
+
+public class Function
+{
+    public async Task FunctionHandler(KinesisEvent evnt, ILambdaContext context)
+    {
+        if (evnt.Records.Count == 0)
+        {
+            context.Logger.LogLine("Empty Kinesis Event received");
+            return;
+        }
+
+        foreach (var record in evnt.Records)
+        {
+            try
+            {
+                context.Logger.LogLine($"Processed Event with EventId: {record.EventId}");
+                string data = await GetRecordDataAsync(record.Kinesis, context);
+                context.Logger.LogLine($"Data: {data}");
+                // TODO: Do interesting work based on the new data
+            }
+            catch (Exception ex)
+            {
+                context.Logger.LogError($"An error occurred {ex.Message}");
+                throw;
```

Comment:
Is this the best practice for handling errors in Kinesis? Because the stream would never move forward right?

Reply:
this is should go to code samples under aws docs. this is another version that handle partial failure